aws/table_aws_spot_price_history.go

```diff
@@ -0,0 +1,151 @@
+	"time"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsSpotPriceHistory(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_spot_price_history",
+		Description: "AWS Spot Price History",
+		List: &plugin.ListConfig{
+			Hydrate: listSpotPriceHistory,
+			KeyColumns: []*plugin.KeyColumn{
+				{Name: "availability_zone", Require: plugin.Optional},
+				{Name: "instance_type", Require: plugin.Optional},
+				{Name: "product_description", Require: plugin.Optional},
+				{Name: "start_time", Require: plugin.Optional},
+				{Name: "end_time", Require: plugin.Optional},
+			},
+		},
+		GetMatrixItemFunc: BuildRegionList,
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{Name: "availability_zone", Description: "The Availability Zone.", Type: proto.ColumnType_STRING},
+			{Name: "instance_type", Description: "The instance type.", Type: proto.ColumnType_STRING},
+			{Name: "product_description", Description: "A general description of the AMI.", Type: proto.ColumnType_STRING},
+			{Name: "spot_price", Description: "The maximum price per unit hour that you are willing to pay for a Spot Instance.", Type: proto.ColumnType_STRING},
+			{Name: "timestamp", Description: "The date and time the request was created", Type: proto.ColumnType_TIMESTAMP},
+			{Name: "start_time", Description: "The date and time, up to the past 90 days, from which to start retrieving the price history data.", Type: proto.ColumnType_TIMESTAMP, Transform: transform.FromQual("start_time")},
+			{Name: "end_time", Description: "The date and time, up to the current date, from which to stop retrieving the price history data.", Type: proto.ColumnType_TIMESTAMP, Transform: transform.FromQual("end_time")},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listSpotPriceHistory(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// Create Session
+	svc, err := EC2Client(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_spot_price_history.listSpotPriceHistory", "connection_error", err)
+		return nil, err
+	}
+	if svc == nil {
+		// Unsupported region, return no data
+		return nil, nil
+	}
+
+	// Limiting the results
+	maxItems := int32(100)
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxItems {
+			if limit < 20 {
+				maxItems = 20
+			} else {
+				maxItems = limit
+			}
+		}
+	}
+
+	input := ec2.DescribeSpotPriceHistoryInput{
+		MaxResults: &maxItems,
+	}
+
+	// use filters instead of availability_zone which doesn't work
+	if d.Quals["availability_zone"] != nil {
+		value := getQualsValueByColumn(d.Quals, "availability_zone", "string").(string)
+		columnName := "availability-zone"
+		input.Filters = append(input.Filters, types.Filter{Name: &columnName, Values: []string{value}})
+	}
+
+	if d.Quals["instance_type"] != nil {
+		value := getQualsValueByColumn(d.Quals, "instance_type", "string")
+		v, ok := value.(string)
+		if !ok {
```

Comment:
Do we need this validation? If the column type is string anyway, wouldn't the query fail earlier if someone passes in a non-string value?

Reply:
I remove all this tests, and i use `d.KeyColumnQuals` now.

At the beginning, I chose `getQualsValueByColumn` because the attribute documentation specifies that this field should not be used for a list call (legacy usage)

Reply:
@aminvielledebatAtBedrock Can you please share a link to the doc/reference where you found that? We can update it if it's outdated or incorrect, thanks!

Reply:
@cbruno10 i was talking about this line https://github.com/turbot/steampipe-plugin-sdk/blame/a027d331d2c6c35d010cc77d98f983b960bea6f7/plugin/query_data.go#L33

The field was renamed in main, but the comment is still the same : 
> this will also be populated for a list call if list key columns are specified however this usage is deprecated and provided for legacy reasons only)

Maybe i'm wrong